core/src/main/scala/org/http4s/client/jdkhttpclient/WebSocketClient.scala

```diff
@@ -0,0 +1,167 @@
+package org.http4s.client.jdkhttpclient
+
+import cats._
+import cats.data.Chain
+import cats.effect._
+import cats.effect.concurrent.{Deferred, Ref, TryableDeferred}
+import cats.implicits._
+import fs2.{Pipe, Pull, Stream}
+import org.http4s.{Headers, Method, Uri}
+import scodec.bits.ByteVector
+
+/**
+  * A websocket request.
+  *
+  * @param uri The URI.
+  * @param headers The headers to send. Some websocket clients reject protocol-specific headers.
+  * @param method The method of the intial HTTP request. Ignored by some clients.
+  * @param subprotocols The subprotocols, descending from most prefered.
+  */
+case class WSRequest(
```

Comment:
Are we reasonably certain no other fields will be added to this?  The builder pattern instead of a case class will promote binary compatibility if this is open to future extension.

Reply:
This API will break when a common websocket api is in http4s(-core). IMHO I don't think bincompat dances are worth it right now.